frontend/src/v5/ui/routes/viewer/drawings/drawingsList/drawingItem/drawingItem.component.tsx

```diff
@@ -30,10 +33,18 @@ type DrawingItemProps = {
 	onClick: React.MouseEventHandler<HTMLDivElement>;
 };
 export const DrawingItem = ({ drawing, onClick }: DrawingItemProps) => {
-	const [latestRevision] = DrawingRevisionsHooksSelectors.selectRevisions(drawing._id);
-	const { calibration, name, drawingNumber, lastUpdated, desc } = drawing;
+	const params = useParams();
+	const history = useHistory();
+	const [, setCalibrationDrawing] = useSearchParam('drawingId');
+	const { calibration, name, drawingNumber, lastUpdated, desc, _id: drawingId } = drawing;
+	const [latestRevision] = DrawingRevisionsHooksSelectors.selectRevisions(drawingId);
 	const { statusCode, revisionCode } = latestRevision || {};
 	const areStatsPending = !revisionCode;
+	
+	const onCalibrateClick = () => {
+		history.push(generatePath(CALIBRATION_VIEWER_ROUTE, params));
+		setCalibrationDrawing(drawingId);
```

Comment:
I can see that this is redirecting twice
<img width="1271" alt="image" src="https://github.com/3drepo/3drepo.io/assets/42032433/befa8b52-a66f-40a0-8966-8f50ee8bc14c">

Can we change it to just change urls once?

Reply:
Don't know if related but when I navigate back instead of going back to the drawings list it goes to containers.